code/game/objects/machinery/floodlight.dm

```diff
@@ -136,6 +136,228 @@
+	idle_power_usage = 50
+	active_power_usage = 2500
+	wrenchable = TRUE
+
+/// Handles the wrench act .
+/obj/machinery/floodlightcombat/wrench_act(mob/living/user, obj/item/I)
+	. = ..()
+	to_chat(user , "<span class='notice'>You begin wrenching [src]'s bolts")
+	playsound( loc, 'sound/items/ratchet.ogg', 60, FALSE)
+	if(!do_after(user, 2 SECONDS, TRUE, src))
+		return FALSE
+	if(anchored)
+		to_chat(user , "<span class='notice'>You unwrench [src]'s bolts")
+		anchored = FALSE
+	else
+		to_chat(user , "<span class='notice'>You wrench down [src]'s bolts")
+		anchored = TRUE
+	set_light(0, 5, "#C5E3E132")
+	on = 0
+
+/// Visually shows that the floodlight has been tipped and breaks all the lights in it.
+/obj/machinery/floodlightcombat/proc/tip_over()
+	for(var/obj/item/light_bulb/tube/T in contents)
+		T.status = 2
+	update_icon()
+	calculate_brightness()
+	var/matrix/A = matrix()
+	A.Turn(90)
+	transform = A
+	density = FALSE
+	tipped = 1
+	on = 0
+
+/// Untip the floodlight
+/obj/machinery/floodlightcombat/proc/flip_back()
+	icon_state = initial(icon_state)
+	density = TRUE
+	var/matrix/A = matrix()
+	transform = A
+	tipped = 0
+
+/obj/machinery/floodlightcombat/Initialize()
+	. = ..()
+	start_processing()
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/machinery/floodlightcombat/LateInitialize()
+	. = ..()
+	power_change()
+
+/obj/machinery/floodlightcombat/process()
+	if(!powered(LIGHT))
+		if(on)
+			on = 0
+			set_light(0, 5, "#C5E3E132")
+			update_icon()
+		return FALSE
+	use_power(on ? active_power_usage : idle_power_usage, LIGHT)
+
+/// Loops between the light tubes until it finds a light to break.
+/obj/machinery/floodlightcombat/proc/break_a_light()
+	var/obj/item/light_bulb/tube/T = pick(contents)
+	if(T.status == 0)
+		T.status = 2
+		calculate_brightness()
+		return TRUE
+	break_a_light()
+
+/obj/machinery/floodlightcombat/attack_alien(mob/living/carbon/xenomorph/M)
+	if(M.a_intent == INTENT_DISARM)
+		to_chat(M, "<span class='xenodanger'>You begin tipping the [src]")
+		if(!density)
+			to_chat(M, "<span class='xenonotice'>The [src] is already tipped over!")
+			return FALSE
+		var/fliptime = 10 SECONDS
+		if(M.mob_size == MOB_SIZE_BIG)
+			fliptime = 5 SECONDS
+		if(isxenocrusher(M))
+			fliptime = 3 SECONDS
+		if(!do_after(M, fliptime, FALSE, src))
+			return FALSE
+		visible_message("<span class='danger'>[M] Flips the [src] , shaterring all the lights!")
+		playsound( loc, 'sound/effects/glasshit.ogg', 60 , FALSE)
+		tip_over()
+		update_icon()
+		return TRUE
+	if(brightness == 0)
+		to_chat(M, "<span class='xenonotice'>There are no lights to slash!")
+		return FALSE
+	else
+		playsound( loc, 'sound/weapons/alien_claw_metal1.ogg', 60, FALSE)
+		to_chat(M, "<span class='xenonotice'>You slash one of the lights!")
+		break_a_light()
+		set_light(brightness, 5, "#C5E3E132")
+		update_icon()
+
+
+/obj/machinery/floodlightcombat/attackby(obj/item/I, mob/user, params)
+	var/list/lights = contents // has to be a static list..
```

Comment:
Can you explain this?

Reply:
The compiler throws a warning about it not being a static list if i don't save the contents somewhere.

Reply:
Please ping me with the error you get
Also also this is hy you should store the lights in a variable on the floodlight so you can just use that without istyping contents